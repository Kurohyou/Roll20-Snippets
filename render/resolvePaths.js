const { resolve } = require('path');
const path = require('path');

const resolvePaths = (source,destination,entry) => {
  const resSource = path.resolve(source,entry.name);
  const resDest = destination ?
    path.resolve(destination,entry.name.replace(/\.pug$/,'.html').replace(/\.scss$/,'.css')) :
    undefined;
  return [resSource,resDest];
}

module.exports = resolvePaths;