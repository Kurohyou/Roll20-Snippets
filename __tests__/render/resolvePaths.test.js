import { describe, it, expect, vi } from 'vitest';
import resolvePaths from './../../render/resolvePaths';

describe('resolvePaths()',()=>{
  it('Should resolve the relative source and destination paths',()=>{
    const retArr = resolvePaths('./','./',{name:'test.pug'});
    expect(retArr).toMatchSnapshot();
  })
});