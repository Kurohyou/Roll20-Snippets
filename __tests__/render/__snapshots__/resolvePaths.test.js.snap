// Vitest Snapshot v1

exports[`resolvePaths() > Should resolve the relative source and destination paths 1`] = `
[
  "C:\\\\Users\\\\scase\\\\OneDrive\\\\Documents\\\\git\\\\kurohyou_studios\\\\Roll20-Snippets\\\\test.pug",
  "C:\\\\Users\\\\scase\\\\OneDrive\\\\Documents\\\\git\\\\kurohyou_studios\\\\Roll20-Snippets\\\\test.html",
]
`;
